### ISSUE TITLE
Question: The training procedure follows Asynchronous gradient descent or Synchronous gradient descent

### ISSUE THREAD

**Issue by PeterSH6** [MITIBMxGraph/SALIENT#1]
Hi @nistath @timkaler ,

Thanks for the great work!
The figure 1 in the paper illustrated the training procedure of SAILENT system.
In the Figure 1, it seems that Batch1 are executed on GPU A and Batch2 are executed on GPU B. 
I wonder if Batch 1 and Batch 2 will form a global batch and they will all_reduce their graident before they update the model parameters? Or Batch 1 and Batch 2 are training asynchrnously?

I'm not sure whether the GPU A and GPU B are working together to perform synchronous graident descent (with world_size=2).
Thanks for your answers!

**Comment by timkaler** [comment]
This code uses PyTorch's DistributedDataParallel module to perform distributed training, and we use it in the standard manner. An all reduce is performed (within the DistributedDataParallel module) to synchronize gradients after each mini-batch --- so for two gpus A and B, GPU A and GPU B are working together to perform synchronous gradient descent. The effective batch size of the training process is N * local_batch_size where N is the number of GPUs participating in training and local_batch_size is the size of the batch processed on each GPU.

**Comment by PeterSH6** [comment]
Thanks for your answers!
